src/core/device.c

```diff
@@ -1048,9 +1048,6 @@ static void device_enumerate(Manager *m) {
                 _cleanup_set_free_ Set *ready_units = NULL, *not_ready_units = NULL;
                 Device *d;
 
-                if (device_is_processed(dev) <= 0)
-                        continue;
```

Comment:
so lemme get this right, ID_PROCESSING= is set when again?

after the main rules block completed a device, but before RUN= is completed. did i get this right? (we probably should add a comment somewher clarifying that).

So that would mean that a device with ID_PROCESSING should probably *not* be considered when a device first shows up, but should be considered if it already was seen before. Right?

but that would suggest that we probably should consider the deseralized state here, shouldn't we?

i.e. if the deserialized state is active, we should not care about the ID_PROCESSING state of the device, but if the deserialized state is unknown or inactive then we should care about it, and skip the device as before, so that it will not be marked active here, until it actually is complete.

Or did i get this wrong? does this make any sense?

Reply:
Everything right, I think. But that is slightly large in post rc2, so here the change is simply reverted.
I'd like to implement exactly what you commented after v257.

Reply:
-> #35351